### PR TITLE
More3.1 ndarray spec

### PR DESCRIPTION
**Pull request by ljulliar**
More NDArray methods and specs (repeat, pad, swapaxes, split, slice, slice_axis, slice_like, take, one_hot)

### CODE REVIEW

**Comment by ljulliar** [comment]
I used a subranch for the PR. It looks like it cumulates the 2 commits. So you may want to forget about the previous PR since this one seem to have both

**Comment by mrkn** [comment]
This week is vacation in Japan, so please wait a moment.
I'm reviewing #23 now.

File: lib/mxnet/ndarray.rb [hunk 1]
```diff
@@ -341,6 +341,31 @@ def swapaxes(*args, **kwargs)
...
+
+    def slice_like(shape_like, *args, **kwargs)
+      Ops.slice_like(self, *args, **kwargs)
+    end
+
+    def take(*args, **kwargs)
+      Ops.take(self, *args, **kwargs)
+    end
+
+    def self.one_hot(indices, depth: nil, on_value: 1.0, off_value: 0.0, dtype: :float32, **kwargs)
```
Comment (mrkn) [inline lib/mxnet/ndarray.rb]: This class method already exists.
Why you redefine it?

**Comment by mrkn** [comment]
@ljulliar I've commented.  Please check it.

**Pull request closed without merging**